Common/Data/Consolidators/VolumeRenkoConsolidator.cs

```diff
@@ -0,0 +1,289 @@
+        {
+            add { _dataConsolidatedHandler += value; }
+            remove { _dataConsolidatedHandler -= value; }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="VolumeRenkoConsolidator"/> class using the specified <paramref name="barSize"/>.
+        /// </summary>
+        /// <param name="barSize">The constant volume size of each bar</param>
+        public VolumeRenkoConsolidator(decimal barSize)
+        {
+            BarSize = barSize;
+        }
+
+        /// <summary>
+        /// Updates this consolidator with the specified data
+        /// </summary>
+        /// <param name="data">The new data for the consolidator</param>
+        public void Update(IBaseData data)
+        {
+            var rate = data.Price;
+            var volume = _volumeLeftOver;
+
+            if (data.GetType() == typeof(QuoteBar))
+            {
+                throw new ArgumentException("VolumeRenkoConsolidator() must be used with TradeBar or Tick data.");
+            }
+            else if (data.GetType() == typeof(Tick))
+            {
+                volume += ((Tick)data).Quantity;
+
+                if (_firstTick)
+                {
+                    _firstTick = false;
+
+                    OpenOn = data.Time;
+                    CloseOn = data.Time;
+                    OpenRate = rate;
+                    HighRate = rate;
+                    LowRate = rate;
+                    CloseRate = rate;
+                }
+                else
+                {
+                    CloseOn = data.Time;
+
+                    if (rate > HighRate)
+                    {
+                        HighRate = rate;
+                    }
+
+                    if (rate < LowRate)
+                    {
+                        LowRate = rate;
+                    }
+
+                    CloseRate = rate;
+                }
+            }
+            else if (data.GetType() == typeof(TradeBar))
+            {
+                volume += ((TradeBar)data).Volume;
+                var open = ((TradeBar)data).Open;
+                var high = ((TradeBar)data).High;
+                var low = ((TradeBar)data).Low;
+                var close = ((TradeBar)data).Close;
+
+                if (_firstTick)
+                {
+                    _firstTick = false;
+
+                    OpenOn = data.Time;
+                    CloseOn = data.EndTime;
+                    OpenRate = open;
+                    HighRate = high;
+                    LowRate = low;
+                    CloseRate = close;
+                }
+                else
+                {
+                    CloseOn = data.EndTime;
+
+                    if (high > HighRate)
+                    {
+                        HighRate = high;
+                    }
+
+                    if (low < LowRate)
+                    {
+                        LowRate = low;
+                    }
```

Comment:
This logic seems like the one in the `Bar.Update` type? `VolumeRenkoBar` could inherit from Bar and reuse a lot I think

Reply:
This is the consolidator class, not the bar class?